package.json

```diff
@@ -36,27 +36,116 @@
     "configuration": {
       "properties": {
         "purple-yolk.cabal.formatter.command": {
-          "default": "cabal-fmt --no-cabal-file --no-tabular",
+          "default": "",
           "description": "The command to run when formatting Cabal files. This should accept an unformatted Cabal file as input on STDIN. It should emit a formatted Cabal file to STDOUT. STDERR will be ignored.",
           "type": "string"
         },
+        "purple-yolk.cabal.formatter.mode": {
+          "default": "discover",
+          "description": "The command to format a Cabal file.",
+          "enum": [
+            "discover",
+            "cabal-fmt",
+            "custom"
+          ],
+          "enumItemLabels": [
+            "Discover",
+            "cabal-fmt",
+            "Custom"
+          ],
+          "markdownEnumDescriptions": [
+            "Automatically determines the appropriate command.",
+            "Uses `cabal-fmt`.",
+            "Uses the command in `#purple-yolk.cabal.formatter.command#`."
+          ],
+          "type": "string"
+        },
         "purple-yolk.haskell.formatter.command": {
-          "default": "ormolu --no-cabal",
+          "default": "",
```

Comment:
This is a noteworthy change. If you previously customized the Haskell formatter command, you'll now have to set the mode to "custom" for that command to be used.